### PR TITLE
Promote editable workspace classes from @internal to @beta

### PR DESCRIPTION
**Pull request by williamkbentley**
Classes were previously @internal, but should be exposed for workspace management apps.
These classes should only be used in a dedicated administer-facing settings manager application. Never in an end-user application.

### CODE REVIEW

**Comment by pmconne** [comment]
Each of the classes you've promoted from `internal` to `beta` (except for `EditableWorkspaceDb`) is an implementation of a `beta` interface. Instead of exposing the implementation details, did you consider adding an interface for editing, and functions for instantiating them?

File: core/backend/src/workspace/Workspace.ts [hunk 1]
```diff
@@ -657,10 +657,11 @@ export class ITwinWorkspaceDb implements WorkspaceDb {
 }
 
 /**
- * An editable [[WorkspaceDb]]. This is used by administrators for creating and modifying `WorkspaceDb`s.
+ * An editable [[WorkspaceDb]]. This is used by administrators for creating and modifying `WorkspaceDb`s and should
+ * only be used in a dedicated "administrative settings manager" application - never in a user facing application.
```
Comment (kabentley) [inline core/backend/src/workspace/Workspace.ts:661]:
```suggestion
 * only be used in a dedicated "administrative workspace editor" application - never in a user facing application.
```

File: core/backend/src/workspace/Workspace.ts [hunk 2]
```diff
@@ -657,10 +657,11 @@ export class ITwinWorkspaceDb implements WorkspaceDb {
...
  * For cloud-backed containers, the write token must be obtained before this class may be used. Only one user at at time
  * may be editing.
- * @internal
+ * @beta
```
Comment (kabentley) [inline core/backend/src/workspace/Workspace.ts:664]: It seems to me that only this class needs to become public, right?

Comment (pmconne) [reply core/backend/src/workspace/Workspace.ts:664]: extract-api is going to fail. You can't give a derived class a higher release tag than its base class.

Comment (kabentley) [reply core/backend/src/workspace/Workspace.ts]: reworked

**Review by pmconne — APPROVED** [review]

**Pull request merged**